src/validation.cpp

```diff
@@ -3860,13 +3860,13 @@ void PruneBlockFilesManual(CChainState& active_chainstate, int nManualPruneHeigh
     }
 }
 
-void CChainState::LoadMempool(const ArgsManager& args)
+void CChainState::LoadMempool(FopenFn mockable_fopen_function)
 {
     if (!m_mempool) return;
-    if (!m_mempool->m_persistence_path.empty()) {
-        ::LoadMempool(*m_mempool, *this);
-    }
-    m_mempool->SetIsLoaded(!ShutdownRequested());
+    auto simple_atmp = [&](const CTransactionRef& tx, int64_t accept_time) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
+        return AcceptToMemoryPool(*this, tx, accept_time, /*bypass_limits=*/false, /*test_accept=*/false).m_result_type == MempoolAcceptResult::ResultType::VALID;
```

Comment:
Do we need all of `AcceptToMemoryPool` here? Without digging too much, I should think we might only need a subset, potentially skipping some of the uncaching/flushing logic?

Reply:
I'm not sure, and I don't know enough about this logic to confidently say. I'd prefer to preserve as much existing behavior as possible for an important module like mempool, which is what I think this does.

Okay to leave as a followup for someone with more knowledge of this logic?

Reply:
Followup entry added in umbrella issue #24303